File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let res = '';
    let i = 0;
    while (i < word1.length && i < word2.length) {
        res += word1[i];
        res += word2[i];
        i++;
    }
    if (i < word1.length) {
        res += word1.slice(i);
    } else res += word2.slice(i);
    
    return res;
};

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = []
        i = 0
        w1, w2 = len(word1), len(word2)

        while i < w1 and i < w2:
            res.append(word1[i])
            res.append(word2[i])
            i += 1
        
        if i < len(word1):
            res.append(word1[i:])
        else:
            res.append(word2[i:])
        
        return "".join(res)